src/variety/country.js

```diff
@@ -7,1062 +7,1129 @@
-					continue;
-				}
-				if (qnumcell.qnum < 0) {
-					continue;
-				}
-				var maxlength = 0;
-				for (var i = 0; i < room.clist.length; i++) {
-					maxlength = Math.max(maxlength, room.clist[i].lpath.clist.length);
-				}
-				if (maxlength < qnumcell.qnum) {
-					this.failcode.add("blLineShort");
-					if (this.checkOnly) {
-						break;
-					}
-					room.clist.seterr(1);
-				}
-			}
+		"FailCode@moonsun": {
+			bkPassTwice: [
+				"線が１つの部屋を２回以上通っています。",
+				"A line passes a room twice or more."
+			],
+			bkNoLine: [
+				"線の通っていない部屋があります。",
+				"A line doesn't pass a room."
+			],
+			bkBothMarksPassed: [
+				"線が月と太陽を両方通っています。",
+				"A line passes both the marks of the moon and the sun."
+			],
+			bkNoMarksPassed: [
+				"線が月も太陽も通っていません。",
+				"A line passes neither the marks of the moon nor the sun."
+			],
+			bkNotAllMSPassed: [
+				"線が全ての月を通っていません。",
+				"A line doesn't pass all of the marks of the moon."
+			],
+			bkNotAllMUPassed: [
+				"線が全ての太陽を通っていません。",
+				"A line doesn't pass all of the marks of the sun."
+			],
+			bkMSPassedGt2: [
+				"月を通った部屋が連続しています。",
+				"A line passes the marks of the moon for two rooms in a row."
+			],
+			bkMUPassedGt2: [
+				"太陽を通った部屋が連続しています。",
+				"A line passes the marks of the sun for two rooms in a row."
+			]
 		},
-		checkLongLineInRoom: function() {
-			var rooms = this.board.roommgr.components;
-			for (var r = 0; r < rooms.length; r++) {
-				var room = rooms[r];
-				var qnumcell = room.clist.getQnumCell();
-				if (qnumcell.isnull) {
-					continue;
-				}
-				if (qnumcell.qnum < 0) {
-					continue;
-				}
-				var visited = room.clist.filter(function(cell) {
-					return cell.lcnt > 0;
-				});
-				for (var c = 0; c < visited.length; c++) {
-					if (visited[c].lpath.clist.length > qnumcell.qnum) {
-						this.failcode.add("blLineLong");
-						visited[c].lpath.clist.seterr(1);
-					}
-				}
-			}
+		"FailCode@onsen": {
+			blPassTwice: [
+				"ある線が１つの部屋を２回以上通っています。",
+				"A line passes a room more than once."
+			],
+			blLineNe: [
+				"線が通過するマスの数が数字と違います。",
+				"The length of the path in a room is different from the number of the loop."
+			],
+			blLineDiff: [
+				"各部屋で線が通過するマスの数が違います。",
+				"The length of the path in a room is different in some rooms."
+			],
+			bkNoLine: ["線の通っていない部屋があります。", "A room remains blank."],
+			lnIsolate: [
+				"線の通っていない○があります。",
+				"A circle doesn't have a line."
+			],
+			lpNumGt2: [
+				"数字が2つ以上含まれたループがあります。",
+				"A loop has more than one number."
+			],
+			lpNoNum: ["○を含んでいないループがあります。", "A loop has no numbers."]
+		},
+		"FailCode@maxi": {
+			blLineShort: [
+				"枠内のどの線も、連続して通るマス数が数字よりも小さいです。",
+				"All lines in a room are shorter than the number."
+			],
+			blLineLong: [
+				"枠内を連続して通るマス数が、数字よりも大きい線があります。",
+				"A line in a room is longer than the number."
+			]
+		},
+		"FailCode@detour": {
+			blWrongTurns: [
+				"The room has the wrong number of turns.",
+				"The room has the wrong number of turns."
```

Comment:
I think for consistency in style this should be "A room has the wrong number of turns."

Reply:
Cool, changed